File: HashMap.cs
public class MyHashMap {
    public class Node {
        public int Key;
        public int Val;
        public Node Next;

        public Node(int key, int val) {
            this.Key = key;
            this.Val = val;
        }
    }

    private Node[] storage;

    public MyHashMap() {
        this.storage = new Node[10000];
    }

    private int HashIdx(int key) {
        return key % 10000;
    }

    private Node Search(Node head, int key) {
        Node prev = head;
        Node curr = head.Next;
        while (curr != null && curr.Key != key) {
            prev = curr;
            curr = curr.Next;
        }
        return prev;
    }

    public void Put(int key, int value) {
        int idx = HashIdx(key);
        if (storage[idx] == null) {
            storage[idx] = new Node(-1, -1); // Dummy head node
        }
        Node prev = Search(storage[idx], key);
        if (prev.Next != null) {
            prev.Next.Val = value; // Update existing key's value
        } else {
            prev.Next = new Node(key, value); // Insert new key-value pair
        }
    }

    public int Get(int key) {
        int idx = HashIdx(key);
        if (storage[idx] == null) return -1; // Key not found
        Node prev = Search(storage[idx], key);
        if (prev.Next == null) return -1; // Key not found
        return prev.Next.Val; // Return the value
    }

    public void Remove(int key) {
        int idx = HashIdx(key);
        if (storage[idx] == null) return; // Key not found
        Node prev = Search(storage[idx], key);
        if (prev.Next == null) return; // Key not found
        prev.Next = prev.Next.Next; // Remove the key-value pair
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.Put(key, value);
 * int param_2 = obj.Get(key);
 * obj.Remove(key);
 */

File: Hashmap.py
class Node:
        def __init__(self, key,val):
            self.key = key
            self.value = val
            self.next = None
class MyHashMap:    

    def __init__(self):
        self.buckets = 1000
        self.storage = [None] * self.buckets
        
    def getBucket(self, key):
        return key % self.buckets

    def find(self, dummy, key):
        bucket = self.getBucket(key)
        prev = dummy
        curr = dummy.next
        while curr and curr.key != key:
            prev = curr
            curr = curr.next
        return prev

    def put(self, key: int, value: int) -> None:
        bucket = self.getBucket(key)
        if not self.storage[bucket]:
            dummy = Node(-1,-1)
            self.storage[bucket] = dummy
            dummy.next = Node(key,value)
        else:
            prev = self.find(self.storage[bucket],key)
            if not prev.next:
                prev.next = Node(key,value)
            else:
                prev.next.value = value

    def get(self, key: int) -> int:
        bucket = self.getBucket(key)
        if not self.storage[bucket]:
            return -1
        else:
            prev = self.find(self.storage[bucket],key)
            if prev.next:
                return prev.next.value
            else:
                return -1

    def remove(self, key: int) -> None:
        bucket = self.getBucket(key)
        if self.storage[bucket]:
            prev = self.find(self.storage[bucket],key)
            if prev.next:
                prev.next = prev.next.next



# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)